src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -6,17 +6,24 @@
 using System.Diagnostics;
 using Microsoft.CodeAnalysis.CSharp.Symbols;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
 using Microsoft.CodeAnalysis.Text;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
     internal partial class Binder
     {
+        /// <summary>
+        /// constants used to designate escaping scopes.
+        /// </summary>
+        internal const uint ExternalScope = 0;
+        internal const uint TopLevelScope = 1;
+
         // Some value kinds are semantically the same and the only distinction is how errors are reported
-        // for those purposes we reserve lowest 2 bits
-        private const int ValueKindInsignificantBits = 2;
-        private const BindValueKind ValueKindSignificantBitsMask = unchecked((BindValueKind)~((1 << ValueKindInsignificantBits)-1));
+        // for those purposes we reserve lowest 3 bits
+        private const int ValueKindInsignificantBits = 3;
+        private const BindValueKind ValueKindSignificantBitsMask = unchecked((BindValueKind)~((1 << ValueKindInsignificantBits) - 1));
```

Comment:
Are we using 3 bits? It looks like the values that distinguish errors are 0-3 only. (We could certainly reserve 3 bits rather than 2, but I'm curious if I'm missing some reason for doing so now.)

Reply:
It appears that we do not. I thought we need another bit. I will revert this part.

---
In reply to: [134881070](https://github.com/dotnet/roslyn/pull/21334#discussion_r134881070) [](ancestors = 134881070)